docs/core/EigenPodManager.md

```diff
@@ -249,148 +282,296 @@ If the Pod Owner is not in undelegation limbo, the added shares are also sent to
+    bool withdrawFundsFromEigenLayer
+) 
+    external 
+    onlyWhenNotPaused(PAUSED_WITHDRAW_RESTAKED_ETH) 
+    onlyNotFrozen(msg.sender) 
     nonReentrant
-    onlyWhenNotPaused(PAUSED_WITHDRAW_RESTAKED_ETH)
 ```
 
+Called by a Staker who is currently in undelegation limbo to exit undelegation limbo and either (i) withdraw beacon chain ETH from EigenLayer, or (ii) stay in EigenLayer and continue restaking.
+
+See *Helpful Definitions* at the top for more info on undelegation limbo.
+
 *Effects*:
-* TODO
+* Staker's undelegation limbo status is removed
+* If withdrawing funds from EigenLayer (funds must already be in `EigenPod`):
+    * Staker's `EigenPod.withdrawableRestakedExecutionLayerGwei` is decreased by their current shares
+    * `EigenPod.withdrawRestakedBeaconChainETH` directly sends `amountWei` to the Staker
+    * Staker's shares are set to 0
+* If not withdrawing funds, see [`DelegationManager.increaseDelegatedShares`](./DelegationManager.md#increasedelegatedshares)
 
 *Requirements*:
-* TODO
+* Pause status MUST NOT be set: `PAUSED_WITHDRAW_RESTAKED_ETH`
+* Caller MUST be a Staker currently in undelegation limbo, and the current block MUST be at least `StrategyManager.withdrawalDelayBlocks` after the limbo's `startBlock`
+* If withdrawing funds from EigenLayer, the funds MUST already be in the Caller's `EigenPod` (and any necessary withdrawals proven via `EigenPod.verifyAndProcessWithdrawals`)
 
 *As of M2*:
+* `slasher.canWithdraw` is a no-op
 * The `onlyNotFrozen` modifier is a no-op
+* The `middlewareTimesIndex` parameter is unused
+
+---
 
-### Other
+### Withdrawing from EigenPodManager
 
-#### `EigenPodManager.forceIntoUndelegationLimbo`
+* [`EigenPodManager.queueWithdrawal`](#eigenpodmanagerqueuewithdrawal)
+* [`EigenPodManager.completeQueuedWithdrawal`](#eigenpodmanagercompletequeuedwithdrawal)
+
+#### `EigenPodManager.queueWithdrawal`
 
 ```solidity
-function forceIntoUndelegationLimbo(
-    address podOwner,
-    address delegatedTo
+function queueWithdrawal(
+    uint256 amountWei,
+    address withdrawer
 )
     external
-    onlyDelegationManager
     onlyWhenNotPaused(PAUSED_WITHDRAW_RESTAKED_ETH)
-    onlyNotFrozen(podOwner)
+    onlyNotFrozen(msg.sender)
     nonReentrant
-    returns (uint256 sharesRemovedFromDelegation)
+    returns (bytes32)
 ```
 
+Called by a Pod Owner to initiate a withdrawal of some amount of their beacon chain ETH. The Pod Owner should first call `verifyAndProcessWithdrawals` to prove that a withdrawal can be made in the first place.
+
+Withdrawals can be completed after a delay via `EigenPodManager.completeQueuedWithdrawal`.
+
 *Effects*:
-* TODO
+* The Pod Owner's share balance is decreased by `amountWei`
+    * If the Pod Owner is not in undelegation limbo, and is delegated to an Operator: (see [`DelegationManager.decreaseDelegatedShares`](./DelegationManager.md#decreasedelegatedshares))
+* The Pod Owner's `EigenPod.withdrawableRestakedExecutionLayerGwei` is decreased by `amountWei`
```

Comment:
if we're referring to this variable by name here, I think we should also mention it (`withdrawableRestakedExecutionLayerGwei`) in the `EigenPod.verifyAndProcessWithdrawals` section

Reply:
ahh actually it's explained below in "requirements". still probably appropriate, but less critical